pkg/state/state_exec_tmpl.go

```diff
@@ -17,8 +17,12 @@ func (st *HelmState) Values() map[string]interface{} {
 }
 
 func (st *HelmState) createReleaseTemplateData(release *ReleaseSpec, vals map[string]interface{}) releaseTemplateData {
+	if release.KubeContext == "" {
+		release.KubeContext = st.HelmDefaults.KubeContext
+	}
```

Comment:
I admit I may have failed to do it universally, but anyway.. it would be nicer if we could make this side-effect-free, so that release isn't modified only by calling createReleaseTemplateData.

```suggestion
	kubeContext := release.KubeContext
	if kubeContext == "" {
		kubeContext = st.HelmDefaults.KubeContext
	}
```

Note that `ApplyOverrides` is ok to modify ReleaseSpec as it's designed as such and the function name states that, too. Also, `ApplyOverrides` overrides by overrids, not defaults. Overriding by defaults is not the thing we've done so far(I believe).

Reply:
Ah, right. I'll fix it, thanks.